homeassistant/components/gpsd/sensor.py

```diff
@@ -43,6 +47,28 @@
 
 DEFAULT_NAME = "GPS"
 
+
+@dataclass(frozen=True, kw_only=True)
+class GpsdSensorDescription(SensorEntityDescription):
+    """Class describing GPSD sensor entities."""
+
+    value_fn: Callable[[AGPS3mechanism], str | None]
+
+
+SENSOR_TYPES: tuple[GpsdSensorDescription, ...] = (
+    GpsdSensorDescription(
+        key="mode",
+        translation_key="mode",
+        name=None,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.ENUM,
+        options=["2d_fix", "3d_fix"],
+        value_fn=lambda agps_thread: {0: None, 2: "2d_fix", 3: "3d_fix"}.get(
+            agps_thread.data_stream.mode, agps_thread.data_stream.mode
+        ),
```

Comment:
```suggestion
        value_fn=lambda agps_thread: {2: "2d_fix", 3: "3d_fix"}.get(
            agps_thread.data_stream.mode
        ),
```
So we make sure only supported options will be returned


Please declare `{2: "2d_fix", 3: "3d_fix"}` as constant

Reply:
Done, thanks!